purchase_allowed_product/tests/test_purchase_allowed_product.py

```diff
@@ -25,76 +25,52 @@ def setUp(self):
             ]
         )
 
-    def new_purchase(self):
-        """Create an empty purchase."""
-        new = self.env["purchase.order"].new()
-        return new
-
-    def new_invoice(self):
-        """Create an empty purchase."""
-        new = self.env["account.invoice"].new()
-        return new
-
-    def set_purchase_partner(self, partner):
-        self.purchase.partner_id = partner
-
-        # It triggers onchange
-        self.purchase.partner_id_change()
-
-        # Ensure the use_only_supplied_product is set
-        self.assertEqual(
-            self.purchase.use_only_supplied_product, partner.use_only_supplied_product
-        )
-
-    def set_invoice_partner(self, partner):
-        self.invoice.partner_id = partner
-
-        # It triggers onchange
-        self.invoice.partner_id_change()
-
-        # Ensure the use_only_supplied_product is set
-        self.assertEqual(
-            self.invoice.use_only_supplied_product, partner.use_only_supplied_product
-        )
-
     def test_purchase_onchange(self):
         """A user creates a purchase from the form."""
         self.partner_4.use_only_supplied_product = True
-
-        with self.env.do_in_onchange():
-            # User presses ``new``
-            self.purchase = self.new_purchase()
-
-            # User changes fields
-            self.set_purchase_partner(self.partner_4)
-
-            self.assertEqual(self.purchase.use_only_supplied_product, True)
-        context = {
-            "restrict_supplier_id": self.purchase.partner_id.id,
-            "use_only_supplied_product": self.purchase.use_only_supplied_product,
-        }
+        with Form(
+            self.env["purchase.order"], view="purchase.purchase_order_form"
+        ) as purchase_form:
+            purchase_form.partner_id = self.partner_4
+            purchase_form.save()
```

Comment:
Non need of this. The `with` automatically saves the form for you. Anyway, you are querying the form itself, not the resultant record.

Reply:
You are rigt. Thanks for the explanation.